test/testdrive/github-5668.td

```diff
@@ -0,0 +1,93 @@
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# For Debezium versions <1.5, Materialize had no way of determining the complete order
+# of messages coming from a Postgres source. (Details in #5668.) We fixed this upstream
+# in Debezium (details in DBZ-2911) by adding a new "sequence" field to the source
+# metadata. This test uses that new sequence field to show that we can now handle messages
+# that seem to go backwards in time (by LSN).
+
+$ set pg-dbz-schema={
+    "type": "record",
+    "name": "envelope",
+    "fields": [
+      {
+        "name": "before",
+        "type": [
+          {
+            "name": "row",
+            "type": "record",
+            "fields": [
+              {"name": "val", "type": "string"}
+            ]
+          },
+          "null"
+        ]
+      },
+      { "name": "after", "type": ["row", "null"] },
+      {
+        "name": "source",
+        "type": {
+          "type": "record",
+          "name": "Source",
+          "namespace": "whatever",
+          "fields": [
+            {
+              "name": "snapshot",
+              "type": [
+                {
+                  "type": "string",
+                  "connect.version": 1,
+                  "connect.parameters": {
+                    "allowed": "true,last,false"
+                  },
+                  "connect.default": "false",
+                  "connect.name": "io.debezium.data.Enum"
+                },
+                "null"
+              ],
+              "default": "false"
+            },
+            {
+              "name": "lsn",
+              "type": ["long", "null"]
+            },
+            {
+              "name": "sequence",
+              "type": ["string", "null"]
+            }
+          ]
+        }
+      }
+    ]
+  }
+
+$ kafka-create-topic topic=pg-dbz-data
+
+$ kafka-ingest format=avro topic=pg-dbz-data schema=${pg-dbz-schema} timestamp=1
+{"before": null, "after": {"row":{"val": "foo"}}, "source": {"lsn": {"long": 3}, "sequence": {"string": "[null, \"3\"]"}, "snapshot": {"string": "false"}}}
+{"before": {"row":{"val": "foo"}}, "after": {"row":{"val": "bar"}}, "source": {"lsn": {"long": 5}, "sequence": {"string": "[null, \"5\"]"}, "snapshot": {"string": "false"}}}
+{"before": {"row":{"val": "bar"}}, "after": {"row":{"val": "baz"}}, "source": {"lsn": {"long": 7}, "sequence": {"string": "[null, \"7\"]"}, "snapshot": {"string": "false"}}}
+
+{"before": null, "after": {"row":{"val": "hello,"}}, "source": {"lsn": {"long": 4}, "sequence": {"string": "[\"8\", \"4\"]"}, "snapshot": {"string": "false"}}}
+{"before": null, "after": {"row":{"val": "world!"}}, "source": {"lsn": {"long": 6}, "sequence": {"string": "[\"8\", \"6\"]"}, "snapshot": {"string": "false"}}}
+
+{"before": null, "after": {"row":{"val": "too late"}}, "source": {"lsn": {"long": 1}, "sequence": {"string": "[null, \"1\"]"}, "snapshot": {"string": "false"}}}
```

Comment:
Why is this this row discarded?

Reply:
So we are now using (`last_commit_lsn`, `lsn`) tuples to deduplicate messages coming from Debezium. Since the `last_commit_lsn` of this row (`null`) is less than the last ingested commit (`8`), it will be skipped.